backend/Application/Aggregates/Contract/Entities/ContractSnapshot.cs

```diff
@@ -0,0 +1,198 @@
+using System.Threading.Tasks;
+using Application.Aggregates.Contract.Types;
+using Application.Api.GraphQL;
+using Application.Api.GraphQL.Transactions;
+using HotChocolate;
+
+namespace Application.Aggregates.Contract.Entities;
+
+/// <summary>
+/// Contains aggregated local for <see cref="Contract"/>.
+/// </summary>
+public sealed class ContractSnapshot
+{
+    public ulong BlockHeight { get; init; }
+    public ulong ContractAddressIndex { get; init; }
+    public ulong ContractAddressSubIndex { get; init; }
+    public string ContractName { get; init; }
+    public string ModuleReference { get; init; }
+    public double Amount { get; init; }
+    
+    [GraphQLIgnore]
+    public ImportSource Source { get; init; }
+    [GraphQLIgnore]
+    public DateTimeOffset CreatedAt { get; init; } = DateTime.UtcNow;
+
+    /// <summary>
+    /// Needed for EF Core
+    /// </summary>
+    private ContractSnapshot()
+    {
+    }
+
+    internal ContractSnapshot(
+        ulong blockHeight,
+        ContractAddress contractAddress,
+        string contractName,
+        string moduleReference,
+        double amount,
+        ImportSource importSource
+        )
+    {
+        BlockHeight = blockHeight;
+        ContractAddressIndex = contractAddress.Index;
+        ContractAddressSubIndex = contractAddress.SubIndex;
+        ContractName = contractName;
+        ModuleReference = moduleReference;
+        Amount = amount;
+        Source = importSource;
+    }
+
+    internal static async Task ImportContractSnapshot(
+        IContractRepositoryFactory repositoryFactory,
+        IContractRepository repository, 
+        ImportSource source
+        )
+    {
+        await using var moduleReadonlyRepository = await repositoryFactory.CreateModuleReadonlyRepository();
+            
+        var contractEventsAdded = repository
+            .GetContractEventsAddedInTransaction()
```

Comment:
Is there some hidden context here? added in which transaction?
How big will this list be?

Reply:
We are currently using entity framework which keeps track of updates entities in the current transaction. In this case there is a transaction for each block height.

The list is already in memory, and will be scoped to data from a single block.